contracts/protocol/modules/v1/MorphoLeverageModule.sol

```diff
@@ -0,0 +1,920 @@
+     * The function reverts if not enough collateral token is redeemed to buy the required minimum amount of borrow tokens.
+     * @param _setToken             Instance of the SetToken
+     * @param _redeemQuantityUnits  Quantity of collateral asset to delever in position units
+     * @param _tradeAdapterName     Name of trade adapter
+     * @param _tradeData            Arbitrary data for trade
+     * @return uint256              Notional repay quantity
+     */
+    function deleverToZeroBorrowBalance(
+        ISetToken _setToken,
+        uint256 _redeemQuantityUnits,
+        string memory _tradeAdapterName,
+        bytes memory _tradeData
+    )
+        external
+        nonReentrant
+        onlyManagerAndValidSet(_setToken)
+        returns (uint256)
+    {
+        IMorpho.MarketParams memory setMarketParams = marketParams[_setToken]; 
+        require(setMarketParams.collateralToken != address(0), "Collateral not set");
+       
+        uint256 setTotalSupply = _setToken.totalSupply();
+        uint256 notionalRedeemQuantity = _redeemQuantityUnits.preciseMul(setTotalSupply);
+        (,uint256 borrowBalance, uint256 borrowShares) = _getCollateralAndBorrowBalances(_setToken, setMarketParams);
+
+        ActionInfo memory deleverInfo = _createAndValidateActionInfoNotional(
+            _setToken,
+            IERC20(setMarketParams.collateralToken),
+            IERC20(setMarketParams.loanToken),
+            notionalRedeemQuantity,
+            borrowBalance,
+            _tradeAdapterName,
+            false,
+            setTotalSupply
+        );
+
+        _withdraw(deleverInfo.setToken, setMarketParams, deleverInfo.notionalSendQuantity);
+
+        _executeTrade(deleverInfo, IERC20(setMarketParams.collateralToken), IERC20(setMarketParams.loanToken), _tradeData);
+
+        _repayBorrow(deleverInfo.setToken, setMarketParams, borrowBalance, borrowShares);
+
+        _sync(deleverInfo.setToken);
+
+        emit LeverageDecreased(
+            _setToken,
+            IERC20(setMarketParams.collateralToken),
+            IERC20(setMarketParams.loanToken),
+            deleverInfo.exchangeAdapter,
+            deleverInfo.notionalSendQuantity,
+            borrowBalance,
+            0   // No protocol fee
+        );
+
+        return borrowBalance;
+    }
+
+    /**
+     * @dev CALLABLE BY ANYBODY: Sync Set positions with ALL enabled Morpho collateral and borrow positions.
+     * @param _setToken               Instance of the SetToken
+     */
+    function sync(ISetToken _setToken) public nonReentrant onlyValidAndInitializedSet(_setToken) {
+        _sync(_setToken);
+    }
+
+    function _sync(ISetToken _setToken) internal {
+        IMorpho.MarketParams memory setMarketParams = marketParams[_setToken];
+        require(setMarketParams.collateralToken != address(0), "Collateral not set");
+        morpho.accrueInterest(setMarketParams);
```

Comment:
It should be called whenever you need updated balances from Morpho

Reply:
Yes exactly. But let's say that you will do an interaction with Morpho in the tx, it might be more gas efficient to accrue the interest first since it will be done anyway during the supply/borrow/repay/withdraw